mast_aladin_lite/table/mast_table.py

```diff
@@ -0,0 +1,271 @@
+from collections import defaultdict
+
+from traitlets import List, Unicode, Bool, Int, Any, observe
+from ipypopout import PopoutButton
+from ipyvuetify import VuetifyTemplate
+from ipywidgets.widgets import widget_serialization
+
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+
+# locations for metadata from MissionMAST:
+missions = ['jwst', 'roman', 'hst']
+table_data_dir = os.path.join(os.path.dirname(__file__), 'data')
+unique_column_path = os.path.join(table_data_dir, 'unique_columns_per_mission.json')
+column_descriptions_path = os.path.join(table_data_dir, 'column_descriptions.json')
+
+# register loaded table widgets as they're initialized
+_table_widgets = dict()
+
+
+def serialize(table):
+    """
+    Convert an astropy table to a list of dictionaries
+    containing each column as a list of pure Python objects.
+    """
+    return [
+        {
+            k: v.tolist()
+            for k, v in dict(row).items()
+        } for row in table
+    ]
+
+
+known_unique_mast_table_cols = [
+    'fileSetName',  # data products from Missions Mast
+    'source_id',    # Gaia
+    'MatchID',      # Hubble Source Catalog
+    'objID',        # PanSTARRS,
+]
+
+
+class MastTable(VuetifyTemplate):
+    """
+    Table widget for observation queries from Mission MAST.
+    """
+    template_file = __file__, "mast_table.vue"
+
+    items = List().tag(sync=True)
+    headers_visible = List().tag(sync=True)
+    headers_avail = List().tag(sync=True)
+    show_if_empty = Bool(True).tag(sync=True)
+    show_rowselect = Bool(True).tag(sync=True)
+    selected_rows = List().tag(sync=True)
+    column_descriptions = List().tag(sync=True)
+    multiselect = Bool(True).tag(sync=True)
+    items_per_page = Int(5).tag(sync=True)
+    show_tooltips = Bool(False).tag(sync=True)
+    menu_open = Bool(False).tag(sync=True)
+    clear_btn_lbl = Unicode('Clear Table').tag(sync=True)
+    popout_button = Any().tag(sync=True, **widget_serialization)
+
+    # item_key is a column of the table with unique values
+    # for each row, enabling selection of the row by lookup
+    item_key = Unicode().tag(sync=True)
+
+    table = None
+    row_select_callbacks = []
+
+    def __init__(self, table, app=None, update_viewport=True, unique_column=None, **kwargs):
+        """
+        Parameters
+        ----------
+        table : `~astropy.table.Table`
+            A table to load.
+
+        app : `~mast_aladin_lite.app.MastAladin`
+            An instance of the ``MastAladin`` app.
+
+        update_viewport : bool (optional, default is `True`)
+            If `True`, set the `~mast_aladin_lite.app.MastAladin`
+            viewport center to the position of the item in the
+            first row of the table on load.
+
+        unique_column : str (optional, default is `None`)
+            A column which contains unique values in each row.
+
+            If no `unique_column` is given, ``MastTable`` will look for a
```

Comment:
I like this approach and am hoping we can use this to identify rows while communicating with Aladin as well.  @cpparts We can figure out later whether that would mean we should refactor to put this capability somewhere else, and whether we want to offer handling for when there are no unique columns.